src/pokemon.c

```diff
@@ -4180,7 +4180,30 @@ bool8 HealStatusConditions(struct Pokemon *mon, u32 healMask, u8 battlerId)
         status &= ~healMask;
         SetMonData(mon, MON_DATA_STATUS, &status);
         if (gMain.inBattle && battlerId != MAX_BATTLERS_COUNT)
+        {
             gBattleMons[battlerId].status1 &= ~healMask;
+            if((healMask & STATUS1_SLEEP))
+            {
+                u32 i = 0;
+                u32 battlerSide = GetBattlerSide(battlerId);
+                struct Pokemon *party;
+
+                if (battlerSide == B_SIDE_PLAYER)
+                    party = gPlayerParty;
+                else
+                    party = gEnemyParty;
```

Comment:
there is a utility function for this in `battle.h`

Reply:
lol this is *all over* the switching code, brb using it in the should switch refactor

Reply:
in this case though isn't doing it explicitly like this faster? cuz `GetBattlerParty` calls `GetBattlerSide` again, and we need to reuse `battlerSide` a few lines further down anyways. Happy to make the replacement if we don't care about doubling up on `GetBattlerSide` calls.

Reply:
i think this works:
```
u32 battlerSide = GetBattlerSide(battlerId);
struct Pokemon *party = GetSideParty(battlerSide);
```
which i think covers Alex's suggestion and only calls `GetBattlerSide` once

Reply:
this is just the same as 
```
struct Pokemon *party = GetSideParty(GetBattlerSide(battlerId))
```
which is the same as 
```
struct Pokemon *party = GetBattlerParty(battlerId)
```

Reply:
but that way we have to call `GetBattlerSide` twice instead of once, cuz the side is referenced again right below

idk if that is any slower, but it is an extra function call

Reply:
true but we are reusing `battlerSide` a few lines below this snippet of code inside of a loop so wed have to call `GetBattlerSide(battlerId)` each time we iterate, hence putting it in a var